File: diamonds-hacks_freefire.py
import discord
from discord.ext import commands

intents = discord.Intents.default()
intents.messages = True

bot = commands.Bot(command_prefix=".", intents=intents)

@bot.event
async def on_ready():
    print("Smarty est en ligne en mode .menu!")

@bot.command(name="menu")
async def menu(ctx, sub_command: str = None):
    if sub_command == "help":
        await send_help(ctx)
    elif sub_command == "diamonds":
        await send_diamonds_menu(ctx)
    else:
        await ctx.send("Commande inconnue. Tapez `.menu help` pour voir la liste des commandes.")

async def send_help(ctx):
    embed = discord.Embed(
        title="Aide de Smarty",
        description="Commandes de Smarty en mode .menu :\n- .menu help : Affiche les commandes disponibles\n- .menu diamonds : Recevez des diamants",
        color=0x00ff00
    )
    await ctx.send(embed=embed)

async def send_diamonds_menu(ctx):
    embed = discord.Embed(
        title="Welcome to diamonds Hack Free Fire!",
        description="Click this button 👇",
        color=0x0000ff
    )

    button = discord.ui.Button(label="Receive diamonds", style=discord.ButtonStyle.primary, custom_id="receive_diamonds")

    view = discord.ui.View()
    view.add_item(button)

    await ctx.send(embed=embed, view=view)

@bot.event
async def on_interaction(interaction):
    if interaction.type == discord.InteractionType.component and interaction.data["custom_id"] == "receive_diamonds":
        await interaction.response.send_message("Suivre les étapes suivant >!")

bot.run("TON_JETON_Ici")

File: diamonds-hacks_freefire.ts
import { Client, Intents, MessageEmbed, MessageActionRow, MessageButton, Interaction } from 'discord.js'

const client = new Client({ intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES] })

client.once('ready', () => {
    console.log('Smarty est en ligne en mode .menu!')
})

client.on('messageCreate', async (message) => {
    if (message.author.bot) return
    const content = message.content.toLowerCase()

    if (content.startsWith('.menu')) {
        const args = content.slice('.menu'.length).trim().split(/ +/)
        const command = args.shift()

        switch (command) {
            case 'help':
                await sendHelp(message)
                break
            case 'diamonds':
                await sendDiamondsMenu(message)
                break
            default:
                message.reply('Commande inconnue. Tapez `.menu help` pour voir la liste des commandes.')
        }
    }
})

async function sendHelp(message: any) {
    const embed = new MessageEmbed()
        .setTitle("Aide de Smarty")
        .setDescription(`
Commandes de Smarty en mode .menu :
- .menu help : Affiche les commandes disponibles
- .menu diamonds : Recevez des diamants
        `)
        .setColor("#00ff00")
    await message.channel.send({ embeds: [embed] })
}

async function sendDiamondsMenu(message: any) {
    const embed = new MessageEmbed()
        .setTitle("Welcome to diamonds Hack Free Fire!")
        .setDescription("Click this button 👇")
        .setColor("#0000ff")

    const button = new MessageActionRow()
        .addComponents(
            new MessageButton()
                .setCustomId('receive_diamonds')
                .setLabel('Receive diamonds')
                .setStyle('PRIMARY')
        )

    await message.channel.send({ embeds: [embed], components: [button] })
}

client.on('interactionCreate', async (interaction: Interaction) => {
    if (!interaction.isButton()) return

    if (interaction.customId === 'receive_diamonds') {
        await interaction.reply("Suivre les étapes suivant >!")
    }
})

client.login('TON_JETON_Ici')
